hypothesis_sqlalchemy/records.py

```diff
@@ -60,10 +60,15 @@ def lists_factory(columns: List[Column],
     def to_unique_fields(row: Tuple[Any, ...]) -> Tuple[Any, ...]:
         return tuple(row[index] for index in unique_indices)
 
+    if unique_indices:
+        unique_by = to_unique_fields
```

Comment:
it is great to return to old code after awhile, now I see that we can replace `to_unique_fields` with [`operator.itemgetter`](https://docs.python.org/3/library/operator.html#operator.itemgetter) like
```python
from operator import itemgetter
...
if unique_indices:
    unique_by = itemgetter(*unique_indices)
else:
    unique_by = None
```

Reply:
this is not quite true for a case with `len(unique_indices) == 1` (`itemgetter` will return just a single object, not a singleton `tuple`), so we can move function definition in the `if` body like
```python
if unique_indices:
    def unique_by(row: Tuple[Any, ...]) -> Tuple[Any, ...]:
        return tuple(row[index] for index in unique_indices)
else:
    unique_by = None
```